File: example/src/main/java/thirtytwo/degrees/halfpipe/example/health/HelloHealthCheck.java
package thirtytwo.degrees.halfpipe.example.health;

import com.yammer.metrics.core.HealthCheck;
import org.springframework.stereotype.Service;

import java.util.Date;

/**
 * User: spencergibb
 * Date: 9/21/12
 * Time: 6:25 PM
 */
@Service
public class HelloHealthCheck extends HealthCheck {

    public HelloHealthCheck() {
        super("hello");
    }

    @Override
    protected Result check() throws Exception {
        System.out.println(new Date() +" works");
        return Result.healthy("The Date class works!");
    }
}

File: scala_2.9.1/src/main/scala/thirtytwo/degrees/halfpipe/scalaexample/health/HelloHealthCheck.scala
package thirtytwo.degrees.halfpipe.scalaexample.health

import org.springframework.stereotype.Service
import com.yammer.metrics.core.HealthCheck
import java.util.Date
import com.yammer.metrics.core.HealthCheck.Result

/**
 * User: spencergibb
 * Date: 9/23/12
 * Time: 12:01 AM
 */
@Service
class HelloHealthCheck extends HealthCheck("hello") {
  def check() = {
    println(new Date()+ " works")
    Result.healthy("The Date class works")
  }
}
